### PR TITLE
QueryHistory: Fix filter strings being regular escaped

### PR DESCRIPTION
**Pull request by svennergr**
**What is this feature?**

I noticed that the FilterInput in QueryHistory is defaulting to escape certain characters. That does not allow to search for strings like `|` or `[`. Since it's a quick fix, I just went with this PR.

**Special notes for your reviewer**:

Test this out by e.g. running Loki queries:
1. Start Grafana
2. `make devenv sources=loki`
3. Configure a Loki DS in Grafana and go to Explore
4. Do e.g. a query like `{job="\"grafana/data\""} |= `f``
5. Open QueryHistory
6. Try to search for `|=`. Not possible without this fix.

### CODE REVIEW

File: public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx [hunk 1]
```diff
@@ -211,6 +211,7 @@ export function RichHistoryQueriesTab(props: Props) {
           )}
           <div className={styles.filterInput}>
             <FilterInput
+              escapeRegex={false}
```
Comment (ifrost) [inline public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx:214]: The same component is used in `RichHistoryStarredTab` so probably also needs to be updated.

It's interesting that this prop is set to `true` by default. @grafana/user-essentials is this component meant to be used in a specific context or mainly when regex is used for search? Should we use different components for filtering? (cc: @grafana/design-system). It's quite easy to forget to set that prop and there might be similar bugs in other places.

I was thinking we could use `<Input prefix={<Icon name="search" />} />` but it doesn't have that nice `X Clear` button :(

Just for context if someone's interested. This component was added to fix https://github.com/grafana/grafana/issues/12972 in https://github.com/grafana/grafana/pull/15382 and the escaping logic was always applied. It was changed to be optional to fix https://github.com/grafana/grafana/issues/55312 in #55314

**Review by ifrost — CHANGES_REQUESTED** [review]
Thanks @svennergr for submitting this! 🙌  Could you just please update `RichHistoryStarredTab` (for the starred tab) because it uses the same component so it has the same bug. The question if we should use this component or something else is on us and we can address it in a separate PR 👍

File: public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx (hunk 1 above)
Comment (svennergr) [reply public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx:214]: Thanks @ifrost. I fixed it in [0648b54](https://github.com/grafana/grafana/pull/64879/commits/0648b5462b090dea1f0086747878199528b373a0).

Btw, do you consider this a bug that should be backported?

Comment (ifrost) [reply public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx:214]: Thanks! It is a bug but I think it should not be backported.

**Review by ifrost — APPROVED** [review]

**Pull request merged**